main/lsp/LSPTypechecker.cc

```diff
@@ -246,28 +246,40 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
         fref.data(*gs).strictLevel = pipeline::decideStrictLevel(*gs, fref, config->opts);
 
         toTypecheck.emplace_back(fref);
+    }
+
+    UnorderedSet<core::FileRef> packageFiles;
 
+    for (auto fref : toTypecheck) {
         // Only need to re-run packager if we're going to delete constants and have to re-define
         // their visibility, which only happens if we're running incrementalNamer.
         if (shouldRunIncrementalNamer) {
             // NOTE: Using `gs` to access package information here assumes that edits to __package.rb
             // files don't take the fast path. We'll want (or maybe need) to revisit this when we start
             // making edits to `__package.rb` take fast paths.
             if (!(fref.data(*gs).isPackage())) {
-                auto pkgName = gs->packageDB().getPackageNameForFile(fref);
-                if (pkgName.exists()) {
+                auto &pkg = gs->packageDB().getPackageForFile(*gs, fref);
+                if (pkg.exists()) {
```

Comment:
AIUI, the check here for `pkgName` before and the check here for `pkg` after are basically identical?  But I assume that it's not, because either this part or the selection process for `__package.rb` files must be the behavior-changing part of the patch.  What am I missing?

Reply:
It's that `toTypecheck` also contains `result.extraFiles`, see line 220 above. Before, we were only looking for `package.rb` files corresponding to `changedFiles`, but now we are also looking for those corresponding to `extraFiles`.

Reply:
Ah, that was the bit I was missing, thank you.